### PR TITLE
EN Endpoint Documentation Update, Corrected documentation for COS Integration and Slack Subscription, COS Integration Read fix , Slack Template crash fix

### PR DESCRIPTION
**Pull request by Divya-Singh1693**
Test:

Screenshot 2024-07-23 at 12 32 51 PM
Screenshot 2024-07-23 at 12 33 07 PM
Screenshot 2024-07-23 at 12 33 42 PM

Metrics and SMTP Allowed IP's Test:
Screenshot 2024-08-09 at 3 54 03 PM

Screenshot 2024-08-09 at 3 54 54 PM

Screenshot 2024-08-09 at 3 55 24 PM

### CODE REVIEW

File: ibm/service/eventnotification/resource_ibm_en_slack_template.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func resourceIBMEnSlackTemplateCreate(context context.Context, d *schema.Resourc
 		options.SetDescription(d.Get("description").(string))
 	}
 
-	params := EmailTemplateParamsMap(d.Get("params").(map[string]interface{}))
+	params := EmailTemplateParamsMap(d.Get("params.0").(map[string]interface{}))
```
Comment (hkantare) [inline ibm/service/eventnotification/resource_ibm_en_slack_template.go:106]: Accessing .0 element will result in crash as params is optional argument

File: go.mod [hunk 2]
```diff
@@ -241,6 +241,7 @@ replace github.com/dgrijalva/jwt-go v3.2.0+incompatible => github.com/golang-jwt
 // add sdk changes.
 replace github.com/portworx/sched-ops v0.0.0-20200831185134-3e8010dc7056 => github.com/portworx/sched-ops v0.20.4-openstorage-rc3 // required by rook v1.7
 
+
```
Comment (hkantare) [inline go.mod:244]: remove this change

**Comment by Divya-Singh1693** [comment]
@hkantare I have updated the slack_template parameter as required parameter.
There are new data source has been added as part of same PR. The smtp_settings has been removed from our SDKs, the alternate is available and have added the same in docs.

**Comment by hkantare** [comment]
Resolve the conflicts

**Comment by Divya-Singh1693** [comment]
@hkantare I have resolved the conflicts. Please review.

**Pull request merged**